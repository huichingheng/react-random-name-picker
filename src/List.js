import React from "react";

const List = props => {
  return (
    <div>
      {props.names.map((name, i) => {
        if (props.luckyWinnerIndex === i) {
          return (
            <li key={i} id="winner">
              {name}
            </li>
          );
        } else {
          return (
            <li key={i} id="">
              {name}
            </li>
          );
        }
      })}
    </div>
  );
};

export default List;
