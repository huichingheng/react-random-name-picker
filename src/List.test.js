import React from "react";
import List from "./List";
import ShallowRender from "react-test-renderer/shallow";

const testData = {
    names: ["gordon", "sahil", "david", "sally", "jane", "alice"],
    luckyWinnerIndex: null,
}

it("renders List component", () => {
  const renderer = new ShallowRender();
  renderer.render(<List
    names={testData.names}
          luckyWinnerIndex={testData.luckyWinnerIndex} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot()
});
