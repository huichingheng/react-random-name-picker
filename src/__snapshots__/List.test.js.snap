// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders List component 1`] = `
<div>
  <li
    id=""
  >
    gordon
  </li>
  <li
    id=""
  >
    sahil
  </li>
  <li
    id=""
  >
    david
  </li>
  <li
    id=""
  >
    sally
  </li>
  <li
    id=""
  >
    jane
  </li>
  <li
    id=""
  >
    alice
  </li>
</div>
`;
